### ISSUE TITLE
Crash upon switching to "Rendered" mode in the 3D viewport after enabling RPR on Linux

### ISSUE THREAD

**Issue by peytondmurray** [GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon#483]
## Issue
Blender crashes when switching to the "Rendered" mode in the 3D viewport if RPR is loaded and the RPR render engine is selected.

## Steps to Reproduce
1. (After the RPR plugin is installed) open blender; default cube appears.
2. From the `Render Properties` tab, select Radeon ProRender from the `Render Engine` dropdown menu.
3. Change the 3D viewport to Render mode by pressing `z` and then clicking on the `Rendered` option

I've attached a scene file, but it's just the default cube with RPR selected as the render engine. To make it crash, I simply change the 3D viewport to "Rendered" mode.

## System Setup (Environment)
**OS Version**: Arch Linux 5.17.1-arch1-1
**Blender Version**: 3.1.2, commit hash `cc66d1020c3b`. Build platform `Linux-5.17.1-arch1-1-x86_64-with-glibc2.35`.
**GPU**: Powercolor 6900 XT, with Mesa 22.0.1 driver
From the "Save System Info" option in blender, here's a little more info about the GPU and driver:
```
renderer:	'AMD Radeon RX 6900 XT (sienna_cichlid, LLVM 13.0.1, DRM 3.44, 5.17.1-arch1-1)'
vendor:		'AMD'
version:	'4.6 (Core Profile) Mesa 22.0.1'
```

## Detailed Description
Launching blender from the console, I get the following when the crash happens:
```
Intrinsic has incorrect argument type!
<4 x i32> (i64, float, <4 x float>, <4 x float>, <4 x float>, <4 x i32>)* @llvm.amdgcn.image.bvh.intersect.ray.i64.v4f32
Error: Linking bitcode failed: linking source & IR libraries.

terminate called after throwing an instance of 'Tahoe::Exception'
  what():  std::exception
Aborted (core dumped)
```
[crash.blend.zip](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/files/8476078/crash.blend.zip)

Let me know if you need any more information!

**Comment by peytondmurray** [comment]
Sure, here it is:
[crash.blend.zip](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/files/8476081/crash.blend.zip)

This _should_ work, I tested unzipping it just now...

**Comment by gkn11** [comment]
I have the same error after upgrading to rocm 5.1.1 (from 5.0.x?). Could be an arch issue.

**Comment by lenhone** [comment]
Same error here also with arch Linux and the [opencl-amd](https://aur.archlinux.org/packages/opencl-amd) package which as of writing uses the precompiled ROCM stuff from https://repo.radeon.com/rocm/apt/5.1.1/pool/main/

**Comment by MinikPLayer** [comment]
Same error here with manjaro

**Comment by lenhone** [comment]
The recently released [v3.3.31](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/releases/tag/v3.3.31) works for me again.

**Comment by peytondmurray** [comment]
Confirmed, updating to v3.3.31 solved the problem. I'm not sure what the problem actually was, but I'll close this as resolved.
